models/users.js

```diff
@@ -397,37 +407,150 @@ const addForVerification = async (userId, discordId, profileImageUrl, discordIma
  * @return {Promise<{message: string}|{message: string}>}
  * @throws {Error} - If error occurs while verifying user's image
  */
-const markAsVerified = async (userId, imageType) => {
+const changePhotoVerificationStatus = async (userId, imageType, status) => {
   try {
-    const verificationUserDataSnapshot = await photoVerificationModel.where("userId", "==", userId).get();
+    const verificationUserDataSnapshot = await photoVerificationModel
+      .where("userId", "==", userId)
+      .where("status", "==", photoVerificationRequestStatus.PENDING)
+      .get();
+
     // THROWS ERROR IF NO DOCUMENT FOUND
     if (verificationUserDataSnapshot.empty) {
       throw new Error("No verification document record data for user was found");
     }
-    // VERIFIES BASED ON THE TYPE OF IMAGE
-    const imageVerificationType = imageType === "discord" ? "discord.approved" : "profile.approved";
+
     const documentRef = verificationUserDataSnapshot.docs[0].ref;
-    await documentRef.update({ [imageVerificationType]: true });
-    return { message: "User image data verified successfully" };
+
+    // if status is rejected then remove the verification entry
+    if (status === photoVerificationRequestStatus.REJECTED) {
+      await documentRef.update({ status });
+      return "User photo verification request rejected successfully";
+    }
+
+    // VERIFIES BASED ON THE TYPE OF IMAGE
+    if (imageType === "both") {
```

Comment:
What is both here?

Reply:
in the photo-verification object, there are two entries, one for profile (RDS profile) and one for discord (discord profile)
the API gives the ability to verify each type one by one, but for the case where you want to update both keys together you pass the imageType as "both", there is also a validadtor to check this,

Reply:
NIT: I think `both` here is confusing